addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+          largeScreenPopoverVisible,
+          containerNode,
+        } = this.getProperties(
+          'smallScreenPopoverVisible',
+          'largeScreenPopoverVisible',
+          'containerNode'
+        );
+
+        if (containerNode) {
+          let containerWidth = containerNode.getBoundingClientRect().width;
+          if (containerWidth > 0) {
+            this.set('containerWidth', containerWidth);
+          }
+        }
+
+        if (smallScreenPopoverVisible || largeScreenPopoverVisible) {
+          this.setProperties({
+            smallScreenPopoverVisible: false,
+            largeScreenPopoverVisible: false,
+          });
+        }
+      },
+      50
+    );
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    let promotedActions = this.get('promotedActions');
+
+    if (promotedActions && promotedActions.length > MAX_PROMOTED_ACTIONS) {
+      warn(
+        `To provide a better user experience. There should only be a maximum of ${MAX_PROMOTED_ACTIONS} promoted actions.`
+      );
+    }
+
+    this.setContainerNode();
+
+    let largeScreenGroupNode = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (largeScreenGroupNode) {
+      this.addEventListener(largeScreenGroupNode, 'resize', this.handleResize);
+    }
+  },
+
+  didRender() {
+    this._super(...arguments);
+
+    let moreActionsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Popover'
+    );
+
+    let largeScreenButtonsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (moreActionsEl) {
+      this.setMoreActionsNode(moreActionsEl);
+    }
+
+    if (largeScreenButtonsEl) {
+      this.setLargeScreenButtonsNode(largeScreenButtonsEl);
+    }
+
+    let {
+      actionsCollection,
+      promotedActions,
+      moreActionsNode,
+      addedMoreActionsWidthForMeasuring,
+      largeScreenButtonsNode,
+      containerNode,
+    } = this.getProperties(
+      'actionsCollection',
+      'promotedActions',
+      'moreActionsNode',
+      'addedMoreActionsWidthForMeasuring',
+      'largeScreenButtonsNode',
+      'containerNode'
+    );
+
+    if (promotedActions && !actionsCollection && moreActionsNode) {
+      addedMoreActionsWidthForMeasuring = moreActionsNode.getBoundingClientRect()
+        .width;
+    }
+
+    let bulkActionsWidth = largeScreenButtonsNode
+      ? largeScreenButtonsNode.getBoundingClientRect().width -
+        addedMoreActionsWidthForMeasuring
+      : 0;
+
+    this.set('bulkActionsWidth', bulkActionsWidth);
+
+    if (containerNode) {
+      this.setProperties({
+        containerWidth: containerNode.getBoundingClientRect().width,
+        measuring: false,
+      });
+    }
+  },
+
+  actions: {
+    toggleSmallScreenPopover() {
+      this.set(
+        'smallScreenPopoverVisible',
+        !this.get('smallScreenPopoverVisible')
+      );
```

Comment:
Currently `polaris-popover` behaves differently to the React implementation in that it's opened/closed using actions and the `ember-basic-dropdown` trigger, so these popover properties/actions won't have the same effect as the React component. Can you look at the React implementation's behaviour and figure out a way to replicate it using `polaris-popover` if possible (assuming they behave differently... if they behave the same already then look at removing any popover-related code from here that doesn't have any effect) 👍

Reply:
Ah right - I can refactor to just use `popover.close` in the `onActionAnyItem` call of the popover, but do you have any thoughts on how we can close the popover in the `resize` handler?

The react implementation just sets `smallScreenPopoverVisible` and `smallScreenPopoverVisible` to `false` on `resize` but we aren't able to take advantage of this and I'm not sure of a good way to pass the currently active `popover` object into the component's methods 🤔

Reply:
Hmm... maybe it's time we ported the [`EventListener`](https://github.com/Shopify/polaris-react/blob/master/src/components/EventListener/EventListener.tsx) component and use that - that way we could maybe pass `popover`/`popover.close` into the action handler, either directly (e.g. `onAction=(action popover.close)`) or indirectly (`onAction=(action "someAction" popover`, and then `someAction` does whatever it needs to do plus calls `popover.close()`) 🤔